### PR TITLE
Solve (partially) dupes/missing rows/wrong sequences in repeat tables

### LINKED ISSUE #134: Dupes or missing rows in repeat tables prevents users from listing and exporting submissions
There are two things that could happen inside a repeat: we have a missing row or we have a dupe row

Aggregate relies on the `_ORDINAL_NUMBER` field which should contain a sequence of ascending ints without holes and without dupes. If this sequence is wrong for 1 submission, we fail and the user can't see or export any submission at all

The safest thing to do is to show only the subset of repeat rows that match the expected sequence (1,2,3,4...) but this would lead to a rather difficult to predict behavior:
- If there is a hole as in `[1, 2, 4, 5, 6, 7]`: the user will only see rows `[1, 2]`
- If there is a dupe in the place of the hole as in `[1, 2, 2, 4, 5, 6, 7]`: the user will see `[1, 2(the first), 4, 5, 6, 7]`

The other option would be to show everything (dupes included). We would use `_ORDINAL_NUMBER` as an ordering parameter only and dupes/missing numbers wouldn't throw an error.

In any case, we wouldn't throw. Instead, we would leave a trace in the logs.

### PR DESCRIPTION
**Pull request by ggalmazor**
Closes #134

This PR solves a problem I introduced in a wrong rebase in PR #141 

Warning the user about missing/dupes in repeat tables would be handled in #135 

To test this PR, you can use these forms: 

- Normal repeat [Untitled Form.xml.tar.gz](https://github.com/opendatakit/aggregate/files/1476411/Untitled.Form.xml.tar.gz)
- Nested repeat [nested-repeats.xml.tar.gz](https://github.com/opendatakit/aggregate/files/1499968/nested-repeats.xml.tar.gz)

Just send a submission and alter the database to produce missing rows and duped rows. This example has a dupe row for `_ORDINAL_NUMBER` 1 and has no row for `_ORDINAL_NUMBER` 2.

_URI | _CREATOR_URI_USER | _CREATION_DATE | _LAST_UPDATE_URI_USER | _LAST_UPDATE_DATE | _PARENT_AURI | _ORDINAL_NUMBER | _TOP_LEVEL_AURI | SOME_TEXT
-- | -- | -- | -- | -- | -- | -- | -- | --
uuid:46666a40-dfbf-4255-bcd4-e34b33325000 | anonymousUser | 2017-11-13 22:16:31.859000 | NULL | 2017-11-15 22:16:31.859000 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 1 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 1
uuid:46666a40-dfbf-4255-bcd4-e34b33325001 | anonymousUser | 2017-11-14 22:16:31.859000 | NULL | 2017-11-15 22:16:31.859000 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 1 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 2
uuid:46666a40-dfbf-4255-bcd4-e34b33325002 | anonymousUser | 2017-11-15 22:16:31.859000 | NULL | 2017-11-15 22:16:31.859000 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 3 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 3

With this data, you should be able to see and export a group with rows for the text '2' and '3':

demo_repeat_group

### CODE REVIEW

**Comment by ggalmazor** [comment]
It doesn't work with nested repeats. I'll look into that tomorrow

**Comment by ggalmazor** [comment]
Alright! Fixed for nested repeat tables too :)

**Comment by lognaturel** [comment]
🎉 @yanokwa that was the only special case I could think of. Any from you?

File: src/main/resources/security.properties [hunk 1]
```diff
@@ -28,7 +28,7 @@ security.server.channelType=REQUIRES_INSECURE_CHANNEL
 # external services.
 #
 # This is configured during install.  If blank, discovers an IP address
-security.server.hostname=localhost
+security.server.hostname=192.168.1.161
```
Comment (yanokwa) [inline src/main/resources/security.properties]: This should be localhost.

**Pull request merged**